md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/AnalyticsAppAPI.java

```diff
@@ -0,0 +1,279 @@
+package com.wipro.ats.bdre.md.rest.ext;
+
+import com.wipro.ats.bdre.exception.MetadataException;
+import com.wipro.ats.bdre.md.beans.table.AnalyticsApps;
+import com.wipro.ats.bdre.md.dao.AnalyticsAppsDAO;
+import com.wipro.ats.bdre.md.dao.ProcessDAO;
+import com.wipro.ats.bdre.md.dao.UserRolesDAO;
+import com.wipro.ats.bdre.md.dao.jpa.*;
+import com.wipro.ats.bdre.md.dao.jpa.Process;
+import com.wipro.ats.bdre.md.dao.jpa.Properties;
+import com.wipro.ats.bdre.md.rest.RestWrapper;
+import com.wipro.ats.bdre.md.rest.util.Dao2TableUtil;
+import com.wipro.ats.bdre.md.rest.util.DateConverter;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.security.Principal;
+import java.util.*;
+
+/**
+ * Created by SU324335 on 4/13/2016.
+ */
+@Controller
+@RequestMapping("/analyticsapp")
+public class AnalyticsAppAPI {
+    private static final Logger LOGGER = Logger.getLogger(AnalyticsAppAPI.class);
+    @Autowired
+    private ProcessDAO processDAO;
+    @Autowired
+    private AnalyticsAppsDAO analyticsAppsDAO;
+    @Autowired
+    UserRolesDAO userRolesDAO;
+
+    @RequestMapping(value = {"/createjobs"}, method = RequestMethod.POST)
+
+    @ResponseBody
+    public RestWrapper createJob(HttpServletRequest request,@RequestParam Map<String, String> map, Principal principal) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) The Cyclomatic Complexity of this method "createJob" is 15 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)